api/modules/budgeting.js

```diff
@@ -1,9 +1,33 @@
 const moment = require('moment')
+const { DEFAULT_STRIPE_CURRENCY } = require('../config/constants')
 
 const budgeting = module.exports = (() => {
     const db = require('../models')
     const clientManagement = require('./clientManagement')
 
+    const checkExistingPaymentsWithExternalId = async (params) => {
+        const existingPayments = await db.models.Payment.findAll({
+            where: {
+                external_uuid: params.id
+            }
+        })
+        
+        if (existingPayments) {
+            existingPayments.forEach(async payment => {
+                const client = await clientManagement.findClientWithId(payment.client_id)
+
+                if (client.currency != DEFAULT_STRIPE_CURRENCY) {
+                    try {
+                        const deletedCustomer = await clientManagement.deleteClientUuid({ id: client.external_uuid })
```

Comment:
Change to `disconnectedCustomer` to remove the confusion of the client being "deleted" when it is actually being updated to clear out the value of the `external_uuid`